6684 dataset pg btn responsive

mheppler opened the pull request:
**What this PR does / why we need it**:

- action btns on dataset and file pg need more scalable space
- primary actions on dataset and file pg need to be more prominent
- responsive layout on dataset and file pg needs to be improved
- need a lil more color on dataset and file pg

**Which issue(s) this PR closes**:

Closes #6684 Layout and styling of action buttons on the dataset and file page
Closes #3815 Improve visibility of buttons for data owners
Closes #1664 Dataset Publish/Submit for Review - Optimize Button/Popup Code
Closes #1221 Deaccessioning: What should show in search card?
Closes #1063 Mobile Dataverse: Solution for Displaying Dataverse on a Mobile Device

Ref #6605 Dataset Page Redesign
Ref #6604 File Page Redesign
Ref #3953 Improve the clarity of page types and their purpose

**Special notes for your reviewer**:

Thank you for your time.

**Suggestions on how to test this**:

@kcondon, I would be happy to meet and discuss what I should put here to help you.

**Does this PR introduce a user interface change?**:

Well, yeah. (Link to mockups: https://sketch.cloud/s/mWGQJ/a/4KZgLZ)

**Is there a release notes update needed for this change?**:

N/A

**Additional documentation**:

N/A

scolapasta commented on src/main/java/propertyFiles/Bundle.properties:
```diff
@@ -1219,7 +1219,18 @@ dataset.guestbookResponse.guestbook.responseTooLong=Please limit response to 255
+dataset.accessBtn.header.download=Download Options
+dataset.accessBtn.header.explore=Explore Options
+dataset.accessBtn.header.compute=Compute Options
+dataset.linkBtn=Link Dataset
```
There was already "dataset.link=Link Dataset" and "dataset.email.datasetContactBtn=Email Dataset Contact" as examples. Are these now gone / no longer used? Should we be cleaning these up?

mheppler replied:
Good catch. Will make sure to review this as part of the final round of code clean up.

mheppler replied:
Reviewed and resolved duplicates. Organized the bundle a bit to group msg's better. Will have an opportunity to revisit the link btn and msg text in issue #6772. There is ample opportunity to review and refactor the other 2,500+ lines of bundle text, as suggested in #4825.

mheppler commented on src/main/java/edu/harvard/iq/dataverse/DatasetPage.java:
```diff
@@ -371,7 +371,7 @@ public String getThumbnailString() {
 
         if (!readOnly) {
-            DatasetThumbnail datasetThumbnail = dataset.getDatasetThumbnail();
+            DatasetThumbnail datasetThumbnail = dataset.getDatasetThumbnail(ImageThumbConverter.DEFAULT_DATASETLOGO_SIZE);
```
Attn @landreev ... there is another issue that represents additional thumbnail improves that might cover this ... Dataset Thumbnail API - Size Parameter #5679.

landreev replied:
I'm not sure I understand the question, but I think the answer is that code is doing the right thing as is.

landreev replied:
I mean, it's not even clear which size to consider the "default", for the datasets in particular, anymore. Since all these different parts of the application now need different size thumbnail, it actually felt confusing to have a size-less method for just one of them. And I don't think it would keep things simpler/cleaner overall.

mheppler replied:
Thank you for the discussion, @landreev and @scolapasta. Any pertinent details of the decisions here should be moved to #5679. Resolving.

mheppler commented:
Oy. Merging the latest from develop is getting hairier and hairier with each attempt. Had to manually review changes from #6920 and #6917. Hopefully that is the last one. Moving to QA...

kcondon commented:
Issues so far:
1. Two cancel "buttons"at bottom of terms page, one button, one link, only one cancel button at top.

mheppler commented:
@kcondon I just submitted a simple fix for the double cancel btns you saw. That looks like it snuck into this branch from develop. The recently merged PR #6907 from @qqmyers switched the cancel btn from a PrimeFaces component to a basic HTML button, but changing the component also required changing the `styleClass` attribute to `class`, and adding a `jsf:` passthrough prefix to the render attribute.

```
<button id="cancelCreate" class="btn btn-link" value="#{bundle.cancel}" onclick="cancelDatasetCreate();return false;" jsf:rendered="#{DatasetPage.editMode == 'CREATE'}">#{bundle.cancel}</button>
```

mheppler commented:
As suggested by @scolapasta, changed `viewParam` for fileId on file pg to OmniFaces component to fix delete. Related to **Upgrade from Glassfish 4.1 #6230**, which is why I did not see it locally. I'm still rockin' GF cause I could not get Payara 5 set up.

mheppler commented:
Apologies. Reverted issue list to last version before my formatting changes. Moved progress checklist from my perspective to own comment.

- [ ] Restrict/publish/unrestrict/publish file shows unlocked lock on new version.
- [ ] Remove static "Download (Placeholder)" from Access Dataset btn dropdown (see kicked can #6118)
- [x] Two cancel "buttons"at bottom of terms page, one button, one link, only one cancel link at top. (issue from develop #6907)
- [x] Delete file from published dataset while on file page, does nothing, exception in logs (This is not due to this branch, break out but check with Gustavo on fix. #6230,)
- [x] Shape and color of draft tag is not in accordance with mockups: is gray rectangle, ~should: blue tag shape~ (Confirmed with Tania, restore original background colors to publication status labels)

kcondon commented:
@mheppler Done with baseline testing, passing back for finishing touches.

mheppler commented:
Removed the "Download (Placeholder)" from the Access Dataset btn dropdown menu. In order to keep the Access Dataset btn from displaying with no options in the dropdown, I needed to add some tool specific render logic to the new `showAccessDatasetButtonGroup` ui:param.

```
<!-- TO-DO #3513 DOWNLOAD ALL LINK -->
<!-- ADDED TOOLS SPECIFIC LOGIC TO DatasetPage.canDownloadFiles() RENDER LOGIC, 
CAN BE REMOVED WHEN DOWNLOAD ALL FILES IS WIRED UP -->
<ui:param name="showAccessDatasetButtonGroup" 
            value="#{(DatasetPage.canDownloadFiles() 
                        and (DatasetPage.datasetExploreTools.size() >= 1 
                        or (DatasetPage.sessionUserAuthenticated and DatasetPage.showComputeButton())))
                        and (!DatasetPage.workingVersion.deaccessioned
                        or (DatasetPage.workingVersion.deaccessioned and DatasetPage.canUpdateDataset()))}"/>
```

Once #6505 and #3513 are resolved, we can remove that additional logic, and wire up the new download option for the new Access Dataset btn.

kcondon commented:
@mheppler passing back to you to comment on the lock state issue so everything is clearly documented as you had requested, thanks.

djbrooke commented:
Thanks @kcondon @mheppler looking forward to discussing any remaining issues next week!

mheppler commented:
@kcondon I do not have all the information needed to replicate whatever lock issue you saw. I have tried repeatedly yesterday and again this morning on my localhost, demo as well as the dv-internal test server you used to initially reported the issue on.

Please confirm here when you have been able to replicate it again yourself on dv-internal, and I will try there again. Until then, I do not know what else I can do to try to resolve this mysterious issue.

kcondon commented:
@mheppler To reproduce you should configure payara5 on your system, use the doi test credentials and set the pidasyncfilecount=-1 as mentioned: curl -X PUT -d -1 http://localhost:8080/api/admin/settings/:PIDAsynchRegFileCount

kcondon commented:
@mheppler OK, have just reproduce it but it is far simpler than originally reported. The problem is that while publishing it displays the currently published version until publish completed, even though you published while on draft version. 
Steps to reproduce:
1. create a dataset with no file but with the following description: THIS IS DATASET VERSION 1! YOU SHOULD NOT SEE THIS WHEN PUBLISHING DATASET VERSION 2!
2. publish
3. edit metadata, change description to read This is dataset version 2. 
4. publish
5. with async mode enabled, see above, the publish lock always appears while publishing.
6. while publishing is taking place (lock is present), you will see dataset version now shows 1.0 and the description shows the one from v1.0. once published, goes to 2.0.

mheppler commented:
@kcondon thank you for the effort to clarify the publish lock issue you saw on your test server. I have been to finally replicate it there, and opened a new issue (with screenshots!).

Hopefully this clears all the final hurdles to green light merging this might fine PR.

kcondon commented:
A couple additional changes were made that need testing:
-Access button:  not displayed if there are no external tools configured that apply to files in the dataset
-Publish button: has sub link to publish that is grayed out when publish/lock is ongoing.

djbrooke commented:
- Access Dataset button should only show if there is at least one Dataset Explore Tool (we will add it back when there's 
- Change text size Data Citation Standards to match Data Citation 
- Items should appear directly under the header in drop down, only show flyout in cases such as Dataset Citation download
- Publish button, do not show dropdown if there's only one thing to do, this will require coordination on display logic (cc: @scolapasta)

mheppler commented:
@sekmiller contributed a bunch of render logic optimization code, moving a lot of the render logic of the publish/submit for review links/buttons from the XHTML to the DatasetPage.java backing bean. There is now only one publish commandLink in the new button UI xhtml code.

@kcondon, with these improvements, I have linked issue **Dataset Publish/Submit for Review - Optimize Button/Popup Code #1664** to be closed when this PR is merged.

djbrooke commented:
Discussed briefly in the design meeting. The steps remaining here are:

- Resolve merge conflict (thanks @sekmiller and @mheppler for working on this, @scolapasta has also been volunteered)
- QA 
- Merge
- Rejoice

mheppler commented:
Fixed (a16adff25e516ba971a97a910e33f2b7f22dd419) the merge conflict/broken dataset pg added in the last update from develop, related to request access javascript that was added by @qqmyers #6943. (Hope that I didn't break anything even more!)

In order to make sure his new code was delivered to both the dataset pg and file pg, I had to duplicate that code block, since we don't have a very elegant way of delivering two different looking buttons to two different pages in the new button UI.

mheppler commented:
OK. Back to QA.

Thank you for the refactored publish btn and popup code, @sekmiller. The old #1664 issue reporting that has been linked to the original PR comment above to close it.

I also had to back out an unnecessary duplication of javascript that I added in my last merge conflict fix and referenced in my last comment above.

kcondon commented:
Issues found:
1. Button color scheme no longer blue as in mockups. This may have been a late stage change.
